salt/state.py

```diff
@@ -787,6 +787,36 @@ def __init__(
         self.inject_globals = {}
         self.mocked = mocked
 
+    def _match_global_state_conditions(self, full, state, name):
+        """
+        Return ``None`` if global state conditions are met. Otherwise, pass a
+        return dictionary which effectively creates a no-op outcome.
+
+        This operation is "explicit allow", in that ANY state and condition
+        combination which matches will allow the state to be run.
+        """
+        matches = []
+        ret = None
+        ret_dict = {
+            "name": name,
+            "comment": "Failed to meet global state conditions. State not called.",
+            "changes": {},
+            "result": None,
+        }
+        if isinstance(self.opts.get("global_state_conditions"), dict):
```

Comment:
```suggestion
        if isinstance(self.functions["config.get"]("global_state_conditions"), dict):
```

@whytewolf , like this?

Reply:
> 

That is closer. I meant `config.option`, I confuse it with `config.get` a few times, which i know is available within the functions dict.  

also i would suggest dumping the output into it's own variable and passing that variable to the if and for loops. just to save the processing of calling config.option multiple times on the same information. i know the calls are minimal but the ocd in me.